source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -0,0 +1,117 @@
+  ipv6_addr.set_addr({address.cbegin(), address.cend()});
+  ipv6_addr.set_scope_id(scope_id);
+  grpc_sock_addr.mutable_ipv6()->CopyFrom(ipv6_addr);
+  return grpc_sock_addr;
+}
+
+void EXPECT_IPV6_ADDR(
+    const SockAddrHolder& converted_addr,
+    pb::uint32 port,
+    pb::uint32 flow_info,
+    const std::vector<char>& address,
+    pb::uint32 scope_id)
+{
+  auto addr_ptr = static_cast<const nxsockaddr*>(converted_addr);
+  auto reinterpreted_ipv6_addr = reinterpret_cast<const nxsockaddr_in6*>(addr_ptr);
+
+  EXPECT_EQ(sizeof(nxsockaddr_in6), converted_addr.size());
+  EXPECT_EQ(nxAF_INET6, addr_ptr->sa_family);
+  EXPECT_EQ(port, reinterpreted_ipv6_addr->sin6_port);
+  EXPECT_EQ(flow_info, reinterpreted_ipv6_addr->sin6_flowinfo);
+  EXPECT_THAT(reinterpreted_ipv6_addr->sin6_addr.addr, ElementsAreArray(address.data(), address.size()));
+  EXPECT_EQ(scope_id, reinterpreted_ipv6_addr->sin6_scope_id);
+}
+
+TEST(XnetConvertersTests, IPv6GrpcSockAddr_ConvertFromGrpc_CreatesIPv6NXSockAddr)
+{
+  constexpr auto PORT = 1234;
+  constexpr auto FLOW_INFO = 9999;
+  const auto ADDRESS = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF};
+  constexpr auto SCOPE_ID = 8888;
+  auto grpc_sock_addr = create_addr_ipv6(PORT, FLOW_INFO, ADDRESS, SCOPE_ID);
+
+  auto converted_addr = convert_from_grpc<nxsockaddr>(grpc_sock_addr);
+
+  EXPECT_IPV6_ADDR(converted_addr, PORT, FLOW_INFO, ADDRESS, SCOPE_ID);
+}
+
+TEST(XnetConvertersTests, IPv6GrpcSockAddrWithTooShortAddress_ConvertFromGrpc_CreatesAddrWithZeroesInMissingBytes)
+{
+  constexpr auto PORT = 1234;
+  constexpr auto FLOW_INFO = 9999;
+  const auto ADDRESS = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9};
+  const auto ADDRESS_WITH_ZEROES = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0};
+  constexpr auto SCOPE_ID = 8888;
+  auto grpc_sock_addr = create_addr_ipv6(PORT, FLOW_INFO, ADDRESS, SCOPE_ID);
+
+  auto converted_addr = convert_from_grpc<nxsockaddr>(grpc_sock_addr);
+
+  EXPECT_IPV6_ADDR(converted_addr, PORT, FLOW_INFO, ADDRESS_WITH_ZEROES, SCOPE_ID);
+}
+
+TEST(XnetConvertersTests, IPv6GrpcSockAddrTooLongAddress_ConvertFromGrpc_CreatesIPv6NXSockAddrWithExtraBytesTruncated)
+{
+  constexpr auto PORT = 1234;
+  constexpr auto FLOW_INFO = 9999;
+  const auto ADDRESS = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF, 0x10, 0x11};
+  const auto TRUNCATED_ADDRESS = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF};
+  constexpr auto SCOPE_ID = 8888;
+  auto grpc_sock_addr = create_addr_ipv6(PORT, FLOW_INFO, ADDRESS, SCOPE_ID);
+
+  auto converted_addr = convert_from_grpc<nxsockaddr>(grpc_sock_addr);
+
+  EXPECT_IPV6_ADDR(converted_addr, PORT, FLOW_INFO, TRUNCATED_ADDRESS, SCOPE_ID);
+}
+
+TEST(XnetConvertersTests, UnsetGrpcSockAddr_ConvertFromGrpc_CreatesZeroedOutSockAddr)
```

Comment:
Maybe I am wrong, but to me "zeroed out" would have elements with the value of zero. Rather than a vector with size of 0

Reply:
Hm, yeah. "Zeroed out" in this context is more in the implementation domain. The struct is literally all zeros:

```c++
std::memset(&addr, 0, sizeof(addr));
```

And the asserts are what that looks like at an API level (empty/unspecified).

I think it's reasonable to be verbose here. "CreatesEmptySockAddrWithUnspecifiedAddressFamily".